tests/storage/blockvolume_test.py

```diff
@@ -196,3 +199,14 @@ def test_repair_capacity(self, capacity, virtual_size, expected_capacity):
 
             vol.updateInvalidatedSize()
             assert vol.getMetadata().capacity == expected_capacity
+
+    @permutations([
+        # format, type (prealloc), expected_result
+        (sc.RAW_FORMAT, sc.PREALLOCATED_VOL, False),  # invalid format and type
+        (sc.COW_FORMAT, sc.PREALLOCATED_VOL, False),  # invalid type
+        (sc.RAW_FORMAT, sc.SPARSE_VOL, False),  # invalid format
+        (sc.COW_FORMAT, sc.SPARSE_VOL, True),  # all OK
+    ])
+    def test_can_reduce(self, format, type, expected_result):
+        with self.make_volume(size=GiB, format=format, prealloc=type) as vol:
```

Comment:
prealloc=type is very strange - if we call the argument prealloc=, we want to call the parameter prealloc.